source/common/grpc/typed_async_client.h

```diff
@@ -35,8 +35,13 @@ template <typename Request> class AsyncStream /* : public RawAsyncStream */ {
 public:
   AsyncStream() = default;
   AsyncStream(RawAsyncStream* stream) : stream_(stream) {}
-  void sendMessage(const Protobuf::Message& request, bool end_stream) {
-    Internal::sendMessageUntyped(stream_, std::move(request), end_stream);
+  bool sendMessage(const Protobuf::Message& request, bool end_stream) {
+    if (stream_ == nullptr) {
+      return false;
+    } else {
```

Comment:
This feels like the wrong solution in this case (one red flag - why isn't the same check being performed for the other methods, e.g., in `resetStream()`)?
Can the AsyncStream be created using the second c'tor?

Reply:
@adisuissa , I check the nullptr in calling part and restore this code.